rust/otel-arrow-rust/src/otap/schema.rs

```diff
@@ -115,10 +115,31 @@ impl SchemaIdBuilder {
             }
 
             Map(field, _) => {
+                let [key_field, value_field] = match field.data_type() {
+                    Struct(fields) => {
+                        // Try the common case first: two fields named "key" and "value"
+                        match fields.iter().as_slice() {
+                            [k, v] if k.name() == "key" && v.name() == "value" => [k, v],
+                            [v, k] if k.name() == "key" && v.name() == "value" => [k, v],
+                            _ => {
+                                let key = fields
+                                    .iter()
+                                    .find(|f| f.name() == "key")
+                                    .expect("Map field must contain a field named 'key'");
+                                let value = fields
+                                    .iter()
+                                    .find(|f| f.name() == "value")
+                                    .expect("Map field must contain a field named 'value'");
+                                [key, value]
+                            }
+                        }
+                    }
+                    _ => panic!("Map field must be a struct"),
```

Comment:
I'm not sure we'll want to panic here on the off chance that the fields do not contain these keys. The arrow-rs map docs state: https://docs.rs/arrow/latest/arrow/datatypes/enum.DataType.html#variant.Map

> In a field with Map type, the field has a child Struct field, which then has two children: key type and the second the value type. The names of the child fields may be respectively “entries”, “key”, and “value”, _but this is not enforced._

Since it's not enforced, I think we'll want to handle this gracefully. Maybe this `match` could return an `Option<[Field, Field]>`, and if the option is `None` we use the existing logic as a fallback. Does that sound OK?

Reply:
Yeah that could work too @utpilla -- it simplifies the code and it's sort of similar where it assumes the key is field[0] and value is field[1]:

https://github.com/apache/arrow-go/blob/3ae84281674622d33b4617c878e099d13d4a1113/arrow/datatype_nested.go#L593-L596